source/common/tracing/http_tracer_impl.cc

```diff
@@ -40,6 +25,20 @@ static std::string valueOrDefault(const Http::HeaderEntry* header, const char* d
   return header ? header->value().c_str() : default_value;
 }
 
+static std::string buildUrl(const Http::HeaderMap& request_headers) {
+  std::string path = request_headers.EnvoyOriginalPath()
+                         ? request_headers.EnvoyOriginalPath()->value().c_str()
+                         : request_headers.Path()->value().c_str();
+  std::string url = fmt::format("{}://{}{}", valueOrDefault(request_headers.Scheme(), ""),
+                                valueOrDefault(request_headers.Host(), ""), path);
+  static const size_t max_path_length = 128;
+  if (url.length() > max_path_length) {
```

Comment:
nit: to be consistent with the truncation logic we are using in access logging for the path, can you apply max_limit to the path component only?